x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,196 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+
+	internalGaugeDecCoins := osmoutils.ConvertCoinsToDecCoins(internalGaugeCoins)
+	externalGaugeDecCoins := osmoutils.ConvertCoinsToDecCoins(externalGaugeCoins)
+	halfOfExternalGaugeDecCoins := osmoutils.ConvertCoinsToDecCoins(halfOfExternalGaugeCoins)
+
+	s.FundAcc(s.TestAccs[0], requiredBalances)
+	s.FundAcc(s.TestAccs[1], requiredBalances)
+	s.FundAcc(s.TestAccs[2], requiredBalances)
+	s.FundModuleAcc(incentivetypes.ModuleName, requiredBalances)
+
+	// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+	clPoolId1 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 1
+	clPoolId2 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 2
+
+	epochInfo := s.App.IncentivesKeeper.GetEpochInfo(s.Ctx)
+
+	// 3. Create external no-lock gauges for CL pools
+	_, clPoolExternalGaugeIdPool1, clPoolExternalGaugeIdPool2 := s.CreateNoLockExternalGauges(clPoolId1.GetId(), clPoolId2.GetId(), externalGaugeCoins)
+
+	// 4. Create Distribution records to incentivize internal CL no-lock gauges
+	s.IncentivizeInternalGauge(clPoolId1.GetId(), clPoolId2.GetId(), epochInfo.Duration, false)
+
+	// 5. let epoch 1 pass
+	// ******************** EPOCH 1 *********************
+	// let 1 epoch pass
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.GetIdentifier(), 1)
+
+	// Validate the results
+	clPoolInternalGaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId1.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 1, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 1, halfOfExternalGaugeCoins)
+
+	internalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	// check internalIncentiveCLPool
+	s.Require().Equal(2, len(internalIncentiveClPool))
+	emissionRateForExternalTokens := sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[0], emissionRateForExternalTokens, internalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[1], emissionRateForExternalTokens, internalIncentiveClPool[1])
+
+	// check externalIncentiveCLPool
+	s.Require().Equal(2, len(externalIncentiveClPool))
+	emissionRateForExternalTokens = sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds() * 2).QuoInt(sdk.NewInt(1000)))
```

Comment:
We should be clear from defining constant / vars and resuing them at gauge creation time and in assertions here instead of hardcoding. This would make it clearer why we expect the results we do.

For example, 
```go
const (
   defaultGaugeValue uint64 = 1_000_000
   numEpochsPaidOverGaugeTwo uint64 = 2
)
externalGaugeCoins := sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(defaultGaugeValue)), sdk.NewCoin("usdc", sdk.NewInt(defaultGaugeValue)))

// Create gauge using `externalGaugeCoins` and `numEpochsPaidOverGaugeTwo`

emissionRateForExternalTokens = sdk.NewDecFromInt(sdk.NewInt(defaultGaugeValue)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds() * numEpochsPaidOverGaugeTwo).QuoInt(sdk.NewInt(1000)))
```